### ISSUE TITLE
Issue with FastAI 2 trt

### ISSUE THREAD

**Issue by rishabh00100** [NVIDIA-AI-IOT/torch2trt#374]
I was trying to convert a model trained on FastAI to tensorRT using this library. I am able to convert the model successfully, but I am not able to replicate the outputs of the model. 

In the native FastAI approach, the output gives me a tensor having values representing confidence values for all the classes trained.
After converting the same model, the output is not a representation of confidence values

Below are the details:

```
import torch
from torch2trt import torch2trt
from fastai.vision import *
from fastai.metrics import error_rate
from PIL import Image
from torchvision import transforms

preprocess = transforms.Compose([
    transforms.Resize(224),
    transforms.CenterCrop(224),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
])

def prediction_time(model, x):
    import time
    times = []
    for i in range(20):
        start_time = time.time()
        y_trt = model(x)

        delta = (time.time() - start_time)
        times.append(delta)
    mean_delta = np.array(times).mean()
    fps = 1/mean_delta
    print('average(sec):{},fps:{}'.format(mean_delta,fps))

# Prepare image for input
# This preprocessing steps are inline with the demo scripts provided in this repo
# But they are how we pass image in the native fastAI model for prediction
input_image = Image.open('test_image/1.jpg')
input_tensor = preprocess(input_image)
input_batch = input_tensor.unsqueeze(0)

# Load the .pkl file using load_learner module in fastAI
# export.pkl is saved inside model_folder
learn = load_learner('model_folder')
learn.model.eval()
model=learn.model

if torch.cuda.is_available():
    input_batch = input_batch.to('cuda')
    model.to('cuda')

# Dummy input for model conversion
x = torch.ones((1, 3, 224, 224)).cuda()
model_trt = torch2trt(learn.model, [x])

x = input_batch
y = model(x)
y_trt = model_trt(x)

prediction_time(model,x)
prediction_time(model_trt,x)
```

When I check the output of both the models (i.e., y and y_trt), I get more or less the similar result which looks something like below:
tensor([[-4.5314,  5.7997]], device='cuda:0', grad_fn=<AddmmBackward>)

But when I run the native fastAI model using the native load code and native preprocessing using the code below:

```
import torch
import fastai
from fastai.vision import *

def process_img_for_prediction(img_path):
	img = PIL.Image.open(img_path).convert("RGB")
	img_tensor = pil2tensor(img, dtype=np.float32).div_(255)
	img = Image(img_tensor)
	return img

learn = load_learner("model_folder")
input = process_img_for_prediction("test_image/1.jpg")
model_output = learn.predict(input)
```

I get the following output
(Category tensor(1), tensor(1), tensor([7.2484e-05, 9.9993e-01]))

As you can see, the FastAI model tensor output is [0.00007, 0.99] while the output from TRT is [-4.5, 5.79] approximately.

Any help in this matter will be highly appreciated as I was able to achieve 4x speed boost with TRT, just the outputs are not aligned.

**Comment by ira7bar** [comment]
I have tried to run softmax on your logits and got the answer you got below:
```
scipy.special.softmax([-4.5314, 5.7997])
>>> [3.26021408e-05 9.99967398e-01]
```
Seems like the results from model(x) (and model_trt) are the model logits, and the results from learn.predict(input) are the logits after softmax.

**Comment by rishabh00100** [comment]
I tested your suggested way on around 200 test images that I had, and I was able to get correct outputs for all of them. The confidence levels are not exact replica, but I think this works for me as of now.

Thank you very much
Have a great day!

**Comment by rishabh00100** [comment]
I had multiple FastAI models. 
The one that I mentioned in the above issue is binary classifier and hence, the softmax layer was able to replicate the output accuracy to acceptable level.

However, when I used the same method to convert another model which was having 8 categories, I was only able to get 57% accuracy. I also had another model which does prediction on 4 categories and on this one, I got 83.5% accuracy.
Both 57% and 83% accuracy are calculated by keeping the original fastAI model output as denominator

I dont think that Softmax is the actual solution to this as I have tried to use the softmax function of scipy (as per your suggestion) and also from the torch library

**Comment by ira7bar** [comment]
You can try to debug the predict function and see what post-processing is done instead of (or in addition to) softmax
